src/tlo/analysis/life_expectancy.py

```diff
@@ -0,0 +1,305 @@
+"""
+This file will read in the output files generated by model runs
+then employs a series of helper functions to
+generate life tables to estimate life expectancy for each run/draw
+
+function: produce_life_expectancy_estimates() will produce summary statistics
+or will output life expectancy at birth for M and F separately for every
+run/draw if argument(...median=False)
+
+"""
+
+from pathlib import Path
+
+import pandas as pd
+
+from tlo import Date
+from tlo.analysis.utils import extract_results, get_scenario_info, load_pickled_dataframes
+
+# define the target period over which we want to estimate life expectancy
+# this should be a one-year period
+TARGET_PERIOD = (Date(2019, 1, 1), Date(2020, 1, 1))
+
+
+# %% --------------------------------------------------------------
+# HELPER FUNCTIONS
+
+
+def _map_age_to_age_group(dataframe):
+    """
+    Maps ages to age-groups in 5-year intervals and adds a new column 'age-group' to the DataFrame.
+
+    Args:
+    - dataframe (pd.DataFrame): The DataFrame containing the age data.
+    - age_column (str): The name of the column containing ages.
+
+    Returns:
+    - pd.DataFrame: The DataFrame with the 'age-group' column added.
+    """
+    # Define age groups in 5-year intervals
+    age_groups = ['0'] + ['1-4'] + [f'{start}-{start + 4}' for start in range(5, 90, 5)] + ['90+']
+
+    # Create a new column 'age-group' based on the age-to-age-group mapping
+    dataframe['age_group'] = pd.cut(dataframe['age'], bins=[0] + [1] + list(range(5, 95, 5)) + [float('inf')],
+                                    labels=age_groups, right=False)
+
+    return dataframe
+
+
+def _create_multi_index_columns(results_folder: Path):
+    info = get_scenario_info(results_folder)
+
+    return pd.MultiIndex.from_product([range(info['number_of_draws']),
+                                       range(info['runs_per_draw'])],
+                                      names=['draw', 'run'])
+
+
+# %% extract key population data for life expectancy calculations
+def _num_deaths_by_age_group(results_folder: Path):
+    """ produces dataframe with mean (+ 95% UI) number of deaths
+    for each draw by age-group
+    dataframe returned: rows=age-gp, columns=draw median, draw lower, draw upper
+    """
+
+    def extract_deaths_by_age_group(df: pd.DataFrame) -> pd.Series:
+        # Call the function to add the 'age-group' column
+        df = _map_age_to_age_group(df)
+        return df.loc[pd.to_datetime(df.date).between(*TARGET_PERIOD)].groupby(["age_group", "sex"])[
+            "person_id"].count()
+
+    return extract_results(
+        results_folder,
+        module="tlo.methods.demography",
+        key="death",
+        custom_generate_series=extract_deaths_by_age_group,
+        do_scaling=False
+    )
+
+
+# get population size in target period
+# get one row of log for each draw/run and compile into df
```

Comment:
move into doctoring

Reply:
This part of the code has been removed and replaced by person-years instead of population size